### PR TITLE
fix: add missing changes for bitbucket cloud SCM provider (#10143)

### LINKED ISSUE #10143: Merge bitbucket cloud SCM generator from previous applicationset repo
The bitbucket cloud SCM provider was added [in this PR](https://github.com/argoproj/applicationset/pull/556/files)
Unfortunately that PR was not included when the applicationset project has been merged into arogcd. 
Would be great to have this feature.

### PR DESCRIPTION
**Pull request by mmerrill3**
This PR completes the original PR from #556 in application set controller repo
Signed-off-by: mmerrill3 <jjpaacks@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [X] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Closes #10143

### CODE REVIEW

File: docs/operator-manual/applicationset/Generators-SCM-Provider.md [hunk 1]
```diff
@@ -219,6 +219,41 @@ spec:
 * `api`: Optional. URL to Azure DevOps. If not set, `https://dev.azure.com` is used.
 * `allBranches`: Optional, default `false`. If `true`, scans every branch of eligible repositories. If `false`, check only the default branch of the eligible repositories.
 
+## Bitbucket
```
Comment (crenshaw-dev) [inline docs/operator-manual/applicationset/Generators-SCM-Provider.md:222]:
```suggestion
## Bitbucket Cloud
```

File: docs/operator-manual/applicationset/Generators-SCM-Provider.md [hunk 2]
```diff
@@ -219,6 +219,41 @@ spec:
...
+metadata:
+  name: myapps
+spec:
+  generators:
+  - scmProvider:
+      bitbucket:
+        # The workspace id (slug).  
+        owner: "mmerrill"
+        # The user to use for basic authentication with an app password.
+        user: "mmerrill"
```
Comment (crenshaw-dev) [inline docs/operator-manual/applicationset/Generators-SCM-Provider.md:238]:
```suggestion
        owner: "example-owner"
        # The user to use for basic authentication with an app password.
        user: "example-user"
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Just a couple tweaks. :-)

**Comment by crenshaw-dev** [comment]
I think this should go all the way back to 2.4?

File: docs/operator-manual/applicationset/Generators-SCM-Provider.md (hunk 2 above)
Comment (mmerrill3) [reply docs/operator-manual/applicationset/Generators-SCM-Provider.md]: Updated and rebased.

**Comment by mmerrill3** [comment]
@crenshaw-dev updated the PR

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**